### DISCUSSION TITLE
codeblock in `rich-text` should include more languages, or plain text at least

### DISCUSSION THREAD

**Discussion by rabelais88** [tinacms/tinacms#3266] [Ideas]
I would like to use more languages in code block: however, TinaCMS's default codeblock only supports very small set of languages: JS, TS, CSS, JSON, HTML, markdown. It should support `Plain Text` at least.

- is TinaCMS rich text editor going to support plain text(or anything that can represent unknown languages) for codeblock in the near future?
- if it is not possible, how can I contribute to make it support more languages?

**Comment by jamespohalloran** [comment]
Hey there! Here is our list of supported languages. 
https://github.com/tinacms/tinacms/blob/main/packages/%40tinacms/toolkit/src/packages/fields/plugins/MdxFieldPlugin/plate/plugins/ui/code-block/index.tsx#L46-L53
I believe they all just need to be supported by Monaco for the syntax highlighting. 

If there's no language supplied, we use "PlainText", so maybe PlainText can be added to our dropdown:
https://github.com/tinacms/tinacms/blob/main/packages/%40tinacms/toolkit/src/packages/fields/plugins/MdxFieldPlugin/plate/plugins/ui/code-block/index.tsx#L285

**Comment by rabelais88** [comment]
@jamespohalloran Thank you. The explanation is very useful. I'll make a quick PR.
